rules/maprule.bzl

```diff
@@ -0,0 +1,26 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Maprule implementation in Starlark.
+
+This module exports the cmd_maprule() and bash_maprule() build rules.
+
+They are the same except for the interpreter they use (cmd.exe and Bash respectively) and thus
+the expected language of their `cmd` attribute. We will refer to them collectively as `maprule`.
```

Comment:
This is a little awkward -- "We will refer to them collectively as 'maprule'. [End of comment]. Perhaps if we want to provide some more context at the file-level in this aliasing file, we should "go the full mile" ?

Reply:
Good point. I replaced that sentence, because the only option I saw for "going the full mile" was to copy the docstrings from maprule_private.bzl, which I didn't want to do.

How does this version look to you?

Reply:
> @c-parsons : does your approval mean I may merge this PR?

Yes. Approval means you can merge :)